charms/jupyter-controller/tests/integration/test_charm.py

```diff
@@ -0,0 +1,43 @@
+import logging
+from pathlib import Path
+
+import yaml
+
+import pytest
+from pytest_operator.plugin import OpsTest
+
+log = logging.getLogger(__name__)
+
+METADATA = yaml.safe_load(Path("./metadata.yaml").read_text())
+CHARM_NAME = METADATA["name"]
+
+
+@pytest.mark.abort_on_fail
+async def test_build_and_deploy(ops_test):
+    await ops_test.model.deploy("istio-pilot", channel="1.5/beta")
+    await ops_test.model.deploy("jupyter-ui")
+    await ops_test.model.add_relation("jupyter-ui", "istio-pilot")
+
+    my_charm = await ops_test.build_charm(".")
+    image_path = METADATA["resources"]["oci-image"]["upstream-source"]
+    resources = {"oci-image": image_path}
+    await ops_test.model.deploy(my_charm, resources=resources)
+    await ops_test.model.wait_for_idle(
+        status="active", raise_on_blocked=False, raise_on_error=False
+    )
+
+
+async def test_status(ops_test):
+    assert ops_test.model.applications[CHARM_NAME].units[0].workload_status == "active"
+
+async def test_integrate_with_prometheus_and_grafana(ops_test:OpsTest):
+    prometheus = "prometheus-k8s"
+    grafana = "grafana-k8s"
+    await ops_test.model.deploy(prometheus, channel="latest/beta")
+    await ops_test.model.deploy(grafana, channel="latest/beta")
+    await ops_test.model.add_relation(prometheus, grafana)
+    await ops_test.model.add_relation(CHARM_NAME, grafana)
+    await ops_test.model.add_relation(prometheus, CHARM_NAME)
+
+    await ops_test.model.wait_for_idle()
```

Comment:
Could we check for some additional things here to make sure the integration is setup correctly?